src/librustc_middle/ty/sty.rs

```diff
@@ -2424,7 +2423,7 @@ impl<'tcx> Const<'tcx> {
                 // when the expression doesn't depend on any parameters.
                 // FIXME(eddyb, skinny121) pass `InferCtxt` into here when it's available, so that
                 // we can call `infcx.const_eval_resolve` which handles inference variables.
-                if substs.has_local_value() {
+                if substs.needs_infer() {
```

Comment:
That's an "e.g." here too. Maybe search for "e.g. inference variables" across the entire codebase?